### PR TITLE
PXP-644 feat: add application init command

### PR DESCRIPTION
**Pull request by mfauzaan**
## Summary
Build the workflow for the wukong application init command to satisfy the specifications. Details about the command inputs and outputs can be found [here](https://mindvalley.atlassian.net/wiki/spaces/PXP/pages/762937345/RFC18+-+Per-Application+Initialisation+Command).

Screenshot 2024-02-02 at 11 35 23 AM

Ticket: https://mindvalley.atlassian.net/browse/PXP-644

### Added 

- [x] Added `wukong application init` command

### CODE REVIEW

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/mindvalley/wukong-cli/pull/192#" title="Author self-approved">mfauzaan</a>*


The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fwukong-cli).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/mindvalley/wukong-cli/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: cli/src/commands/application/init.rs [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+    let mut application_configs = ApplicationConfigs::new()?;
+
+    let name = Text::new("Application name")
+        .with_render_config(get_inquire_render_config())
+        .with_validator(required!("Application name is required"))
+        .with_placeholder("my-first-application")
+        .prompt()?;
+
+    let workflows = get_workflows_from_current_dir()?;
+    let excluded_workflows = inquire::MultiSelect::new("Workflows to exclude", workflows)
```
Comment (onimsha) [inline cli/src/commands/application/init.rs:27]:
```suggestion
    let excluded_workflows = inquire::MultiSelect::new("Workflows to exclude from the Wukong CLI & TUI", workflows)
```

File: cli/src/commands/application/init.rs [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+        .with_render_config(get_inquire_render_config())
+        .with_help_message(
+            "Leave blank to ignore, ↑↓ to move, space to select one, → to all, ← to none",
+        )
+        .prompt()?;
+
+    let mut namespaces: Vec<ApplicationNamespaceConfig> = Vec::new();
+    namespaces.push(configure_namespace("prod".to_string())?);
+
+    let addons = ["Elixir livebook"];
```
Comment (onimsha) [inline cli/src/commands/application/init.rs:37]:
```suggestion
    let addons = ["Elixir Livebook"];
```

File: cli/src/commands/application/init.rs [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+    let selected_addons = inquire::MultiSelect::new("Addons", addons.to_vec())
+        .with_render_config(get_inquire_render_config())
+        .with_help_message(
+            "Leave blank to ignore, ↑↓ to move, space to select one, → to all, ← to none",
+        )
+        .prompt()?;
+
+    println!();
+
+    let configure_staging_namespace = inquire::Confirm::new("Configure the staging namespace?")
```
Comment (onimsha) [inline cli/src/commands/application/init.rs:47]:
```suggestion
    let configure_staging_namespace = inquire::Confirm::new("Do you want to configure the staging namespace?")
```

File: cli/src/commands/application/init.rs [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+                Some(ApplicationAddonElixirLivebookConfig {
+                    enable: true,
+                    allowed_admins: vec![],
+                })
+            },
+        }),
+    });
+
+    let updated_application_configs =
+        inquire::Editor::new("Do you want to review the .wukong.toml file?")
```
Comment (onimsha) [inline cli/src/commands/application/init.rs:83]:
```suggestion
        inquire::Editor::new("Do you want to review the .wukong.toml file before writing to disk ?")
```

File: cli/src/commands/application/init.rs [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+}
+
+fn configure_namespace(namespace_type: String) -> Result<ApplicationNamespaceConfig, WKCliError> {
+    let rollup_strategy_options = ["Rolling Upgrade", "Blue/Green", "Canary"];
+    let rollout_strategy =
+        inquire::Select::new("Rollup strategy", rollup_strategy_options.to_vec())
+            .with_render_config(get_inquire_render_config())
+            .prompt()?;
+
+    let base_replica = CustomType::<u32>::new("Baseline Replicas")
```
Comment (onimsha) [inline cli/src/commands/application/init.rs:129]:
```suggestion
    let base_replica = CustomType::<u32>::new("Number of replicas")
```

File: cli/src/commands/application/init.rs [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+        .with_help_message("Leave it blank to disable AppSignal integration")
+        .prompt()?;
+
+    let honeycomb_dataset = inquire::Text::new("Honeycomb Dataset")
+        .with_render_config(get_inquire_render_config())
+        .with_placeholder(" Optional")
+        .with_help_message("Leave it blank to disable Honeycomb integration")
+        .prompt()?;
+
+    let cloudsql_project_id = inquire::Text::new("Google CloudSQL Project")
```
Comment (onimsha) [inline cli/src/commands/application/init.rs:147]:
```suggestion
    let cloudsql_project_id = inquire::Text::new("Google Project of your CloudSQL instance(s)")
```

**Review by onimsha — APPROVED** [review]

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/mindvalley/wukong-cli/pull/192#" title="Author self-approved">mfauzaan</a>*, *<a href="https://github.com/mindvalley/wukong-cli/pull/192#pullrequestreview-1861853578" title="Approved">onimsha</a>*


The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fwukong-cli).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/mindvalley/wukong-cli/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**